src/transactions/CategoryFilter.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import { PropTypes } from 'prop-types';
+import { useSelector } from 'react-redux';
+
+/**
+ * Displays a select dropdown of categories as options to filter by and
+ * a clear button to reset the dropdown.
+ *
+ * @component
+ */
+function CategoryFilter({ handleFilter }) {
+  const [selected, setSelected] = useState({ selected: 'Filter by Category' });
+  const onSelectChanged = ({ target: { value } }) => {
+    handleFilter({ categoryId: value });
+    setSelected({ selected: value });
```

Comment:
Interesting solution! But storing this same data in state in two places could lead to issues. Instead of making a new piece of state to store basically the same thing, lets pass down the categoryId from TransactionFilters to CategoryFilter through props, and use that to handle keeping the select set to the correct value.